### PR TITLE
new plugin: scan

### PR DESCRIPTION
**Pull request by shiney-wh**
This plugin let you scan opened ports from the remote machine.

### CODE REVIEW

**Comment by shiney-wh** [comment]
Ahah i just read your todo list and you expect some network plugin too x)

File: plugins/network/scan/plugin.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+from api import server
+from api import environ
+import json
+
+if len(plugin.argv) < 2:
+    sys.exit(plugin.help)
+
+
+# Load port -> service database
+with open(os.path.dirname(__file__) + '/ports.json') as database_ports:
```
Comment (nil0x42) [inline plugins/network/scan/plugin.py]: not OS agnostic, prefer use of:
    os.path.join(plugin.path, "ports.json")

File: plugins/network/scan/scanner.php [hunk 2]
```diff
@@ -0,0 +1,14 @@
+<?php
+
+
+$ip = $PHPSPLOIT['IP'];
+$ports = array();
+for($i = $PHPSPLOIT['PORT_MIN']; $i < $PHPSPLOIT['PORT_MAX']; $i++) {
+    if(@fsockopen( $ip, $i, $errstr, $errno, 0.2 )) {
```
Comment (nil0x42) [inline plugins/network/scan/scanner.php]: might be cool to be able to set custom socket timeout on a future release

**Comment by sohelzerdoumi** [comment]
Or we good with this ? 
I will share you a proxy plugin soon.
This new plugin let you open a local proxy port and everythink is sent from the remote server. ;)

**Comment by shiney-wh** [comment]
Are we good with this ? 
I will share you a proxy plugin soon.
This new plugin let you open a local proxy port and everythink is sent from the remote server. ;)

**Comment by nil0x42** [comment]
i ll merge it later, a nice job you did here :)

therefore, i ll probably rename the plugin name into `portscan`

**Comment by nil0x42** [comment]
i d like to discuss with you about what kind of interface you planed for the proxy plugin, and the name of the plugin.
for example, if your proxy is a pivot for HTTP protocol, it might be interesting to name it `httppivot`, or `pivot` if it handles any tcp connection.

we must also think about the best way to interact with it.
for example, it might be interesting to create the pivot as an independent executable, or to be run in a thread, in order to not block access to the phpsploit interface, feel free to discuss with me, i m always @freenode ;)

**Comment by shiney-wh** [comment]
I agree with you, this plugin can be call httppivot.
Yes, i tought about creating a thread for manage it. Like a service.

> httppivot start 3128
> httppivot status
> httppivot stop

**Comment by nil0x42** [comment]
did some changes to `scan` plugin (now `portscan`):

### commit fed6d78096839bc0934bfdc73119160706228ff7
scan plugin: improve interface and stealth
    interface:
        improve display, with a basic nmap-like flavour
    stealth:
        close sockets after opening
        shuffle port scan order

### commit 75864869fd7a11921bc7088e790d78852f184002
scan plugin: add argument checks
    
### commit 0566d1c889979c6dcda25af6ae1258b1d33a651b
rename `scan` plugin into `portscan`

**Pull request merged**